gaia/geo/geo_inputs.py

```diff
@@ -215,10 +216,22 @@ class RasterFileIO(FileIO):
     #: Default output format
     default_output = formats.RASTER
 
-    def read(self, epsg=None):
+    def read(self, as_numpy_array=False, as_single_band=True,
+             old_nodata=None, new_nodata=None, epsg=None):
         """
         Read data from a raster dataset
 
+        :param as_numpy_array: Output data as numpy
+        (default is False i.e. raster osgeo.gdal.Dataset)
+        :param as_single_band: Output data as 2D array of its first band
+        (default is True). If False, returns full 3D array.
+        :param old_nodata: Explicitly identify existing NoData values
+        (default None). If None, attempts to get existing NoData values stored
+        in the raster band.
+        :param new_nodata: Replace NoData values in each band with new_nodata
```

Comment:
Yeah I wasn't quite sure about the best way to logically to describe it. A sketch of my thoughts below:

- A user can try to get the original NoData value stored in raster band `b` by passing `old_nodata=None`.
- However, for a given raster band `b`, `b.GetNoDataValue()` doesn't always return a NoData value. Sometimes raster bands don't explicitly identify a NoData value, in which case `b.GetNoDataValue()` returns `None`.
- If the user knows in advance that a given value `d` is the NoData value within the band, then the user can help the `raster_to_numpy_ar` function by explicitly identifying that value. When a user does explicitly identify `d` as the NoData value then it overrides `b.GetNoDataValue()`, even if `b.GetNoDataValue()` exists (isn't `None`).

Does that make a bit more sense? Let me know if you have any thoughts on rephrasing the description in the docs.